src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -264,6 +276,61 @@ public void Write(byte[] data)
 
         #region Private Methods
 
+        private UnicastIPAddressInformationCollection GetLocalNIAddress(string rni)
+        {
+            foreach (NetworkInterface ni in NetworkInterface.GetAllNetworkInterfaces()) {
+                if (ni.Name == rni) {
+                    return ni.GetIPProperties().UnicastAddresses;
+                }
+            }
+
+            return null;
+        }
+
+        private TcpClient CreateClientFromNetworkInterface()
+        {
+            UnicastIPAddressInformationCollection localAddresses;
+            IPAddress addr, hostAddr;
+            bool isRemoteHostIP = false;
+            bool? isRemoteHostAddrIPv6 = null;
+
+            //Input NI can be IPAddress string
+            if (IPAddress.TryParse(_options.NetworkInterface, out addr)) {
+                var localEndPoint = new IPEndPoint(addr, 0);
+                return new TcpClient(localEndPoint);
+            } else {
+                //Get UnicastIPAddressInformationCollection from the NI adapter
+                localAddresses = GetLocalNIAddress(_options.NetworkInterface);
+            }
+
+            //Throw if input adapter does not exist
+            if (localAddresses == null) {
+                WriteLog.To.Sync.I(Tag, $"Unknown Network Interface {_options.NetworkInterface}.");
+                throw new CouchbaseNetworkException(C4NetworkErrorCode.UnknownHost);
+            }
+
+            // If host is IP, use the family from the IP.
+            isRemoteHostIP = IPAddress.TryParse(_logic.UrlRequest.Host, out hostAddr);
+            if (isRemoteHostIP)
+                isRemoteHostAddrIPv6 = hostAddr.AddressFamily == AddressFamily.InterNetworkV6;
+
+            try {
+                //Create tcp client with ipv6 address for adapter
+                var ipv6 = localAddresses.FirstOrDefault(x => x.Address.AddressFamily == AddressFamily.InterNetworkV6)?.Address;
+                return new TcpClient(new IPEndPoint(ipv6, 0));
```

Comment:
This is not a good enough check.  There are three cases:  

1. Remote Host is an IPv6 address (handled) 
2. Remote Host is a DNS name (handled)
3. Remote Host is an IPv4 address (not handled)

The IP address checks should be done first.  Handle both IPv4 and IPv6 cases, and then fall back in the case of DNS name to either IPv4 or IPv6 (IPv4 safer).